### PR TITLE
filter is added for additional details

### PR DESCRIPTION
**Pull request by KudumulaGangaNandini**
85 coverage
no lints

### CODE REVIEW

File: src/main/java/com/techsophy/tsf/account/dto/RequestProperties.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+@Data
+public class RequestProperties {
+    private String username;
+    private String firstName;
+    private String lastName;
+    private String mobileNumber;
+    private String emailId;
+    private String department;
+    private String userId;
+    private String context;
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/dto/RequestProperties.java]: this is not String context

File: src/main/java/com/techsophy/tsf/account/service/impl/ACLServiceImpl.java [hunk 2]
```diff
@@ -99,18 +101,32 @@ public ACLSchema getACLById(String id)
...
     {
         Map<String,Object> userDetailsFromKeycloak= tokenUtils.getUserInformationMap(tokenUtils.getTokenFromContext());
+        AuditableData auditableData = userFormDataService.getUserFormDataByUserId(userDetailsFromKeycloak.get("userId")+"",false);
+        UserFormDataSchema userFormDataSchema = objectMapper.convertValue(auditableData,UserFormDataSchema.class);
+        RequestProperties requestProperties = new RequestProperties();
+        requestProperties.setUserInfo(userDetailsFromKeycloak);
+        requestProperties.setContext(checkACLSchema.getContext()+"");
+        if(userFormDataSchema!=null&&userFormDataSchema.getUserData()!=null) {
+        Map<String,Object> userData = userFormDataSchema.getUserData();
+        requestProperties.setUserId(userFormDataSchema.getUserId());
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/service/impl/ACLServiceImpl.java:111]: see if you can use https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/index.html?org/apache/commons/lang3/StringUtils.html

File: src/main/java/com/techsophy/tsf/account/service/impl/ACLServiceImpl.java [hunk 3]
```diff
@@ -99,18 +101,32 @@ public ACLSchema getACLById(String id)
     public ACLValidate checkACLAccess(String id, CheckACLSchema checkACLSchema) throws JsonProcessingException, AccessDeniedException
     {
         Map<String,Object> userDetailsFromKeycloak= tokenUtils.getUserInformationMap(tokenUtils.getTokenFromContext());
+        AuditableData auditableData = userFormDataService.getUserFormDataByUserId(userDetailsFromKeycloak.get("userId")+"",false);
+        UserFormDataSchema userFormDataSchema = objectMapper.convertValue(auditableData,UserFormDataSchema.class);
+        RequestProperties requestProperties = new RequestProperties();
+        requestProperties.setUserInfo(userDetailsFromKeycloak);
+        requestProperties.setContext(checkACLSchema.getContext()+"");
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/service/impl/ACLServiceImpl.java]: please test before committing and pushing the code, this is a string here and a map there

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

File: src/main/java/com/techsophy/tsf/account/dto/ACLEntry.java [hunk 4]
```diff
@@ -12,8 +18,26 @@
...
+
+        return  ruleType.match(this.data, request.getUserInfo(),context)?new ACLDecision(decision, renderAdditionalDetails(request)):null;
+    }
+
+
+    private Map<String, Object> renderAdditionalDetails(RequestProperties request)
     {
-        return  ruleType.match(this.data,userDetailsFromKeycloak,context)?new ACLDecision(decision,null):null;
+        ObjectMapper mapper = new ObjectMapper();
+        ST detailTemplate = new ST(new JSONObject(this.additionalDetails).toString(), '<', '>');
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/dto/ACLEntry.java:33]: '<' and '>' should not be hardcoded keep them as configurable with defaults

File: src/main/java/com/techsophy/tsf/account/dto/ACLEntry.java [hunk 5]
```diff
@@ -12,8 +18,26 @@
...
+    {
+
+        return  ruleType.match(this.data, request.getUserInfo(),context)?new ACLDecision(decision, renderAdditionalDetails(request)):null;
+    }
+
+
+    private Map<String, Object> renderAdditionalDetails(RequestProperties request)
     {
-        return  ruleType.match(this.data,userDetailsFromKeycloak,context)?new ACLDecision(decision,null):null;
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/dto/ACLEntry.java:32]: please add test cases to test this logic of replacement

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

File: src/main/java/com/techsophy/tsf/account/dto/ACLEntry.java (hunk 4 above)
Comment (VaibhavTechsophy) [reply src/main/java/com/techsophy/tsf/account/dto/ACLEntry.java:33]: this is still pending

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

File: src/main/java/com/techsophy/tsf/account/repository/document/UserFormDataDefinitionCustomRepositoryImpl.java [hunk 6]
```diff
@@ -33,11 +34,11 @@ public List<UserFormDataDefinition> findByNameOrId(String idOrNameLike)
     }
 
     @Override
-    public UserFormDataDefinition findByUserName(String userName)
+    public Optional<UserFormDataDefinition> findByUserName(String userName)
```
[the remark this answers is not in the document]
Comment (KudumulaGangaNandini) [reply src/main/java/com/techsophy/tsf/account/repository/document/UserFormDataDefinitionCustomRepositoryImpl.java:37]: for exception purpose if with  userName data is not present

**Review by VaibhavTechsophy — APPROVED** [review]

**Pull request merged**